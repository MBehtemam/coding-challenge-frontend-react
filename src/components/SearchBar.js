import styled from 'styled-components';

const SearchBar = styled.section`
  background-color: #950740;
  height: 5vh;
`;

export default SearchBar;
