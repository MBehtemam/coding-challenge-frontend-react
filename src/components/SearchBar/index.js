import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import QueryInput from './components/QueryInput';
import * as QueryActions from '../../logic/Actions/queryActions';

const Wrapper = styled.section`
  display: flex;
  min-height: 100px;
`;

const SearchBar = ({ setQuery, query }) => (
  <Wrapper>
    <QueryInput onBlur={e => setQuery(e.target.value)} defaultValue={query} />
  </Wrapper>
);
const mapStateToProps = state => ({
  query: state.query
});
const mapDispatchToProps = dispatch => ({
  setQuery: query => dispatch(QueryActions.setQuery(query))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBar);

SearchBar.propTypes = {
  setQuery: PropTypes.func,
  query: PropTypes.string
};
SearchBar.defaultProps = {
  query: '',
  setQuery: QueryActions.setQuery
};
