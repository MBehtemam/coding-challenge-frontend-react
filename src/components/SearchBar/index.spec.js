import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import SearchBar from '.';

describe('Test suits for Searchbar ', () => {
  it('should match with snapshot', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
