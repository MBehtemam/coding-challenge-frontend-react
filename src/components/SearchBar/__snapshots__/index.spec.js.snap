// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Searchbar  should match with snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 100px;
}

<section
  className="c0"
>
  <span>
    Search bar section
  </span>
</section>
`;
